File: src/3005.count-elements-with-maximum-frequency.3242/solution.ts
function maxFrequencyElements (nums: number[]): number {
    const frequencies = new Array<number>(101).fill(0);
    let maxFre = 0;
    let maxFreCount = 0;
    for (const n of nums) {
        frequencies[n]++;
        if (frequencies[n] > maxFre) {
            maxFre = frequencies[n];
            maxFreCount = 1;
        } else if (frequencies[n] === maxFre) {
            maxFreCount++;
        }
    }
    return maxFre * maxFreCount;
}

File: src/3005.count-elements-with-maximum-frequency.3242/solution.java
class Solution {
    public int maxFrequencyElements(int[] nums) {
        int[] counts = new int[101];
        for(int n:nums){
            counts[n]++;
        }
        int max = 0;
        int result = 0;
        for(int n:counts){
            if(n>max){
                max = n;
                result = n;
            }else if(n == max){
                result += n;
            }
        }
        return result;
    }
}